client/src/components/formHelpers/FormFields.js

```diff
@@ -33,46 +40,72 @@ function FormFields(props) {
     types = {}
   } = options;
 
+  const nullOrWarning = value => {
+    const error = composeValidators(editorValidator, localhostValidator)(value);
+    return error ? (
+      <HelpBlock>
+        <Alert bsStyle='info'>{error}</Alert>
+      </HelpBlock>
+    ) : null;
+  };
   return (
-    <div>
-      {formFields
-        .filter(formField => !ignored.includes(formField.name))
-        .map(({ name, label }) => (
-          <Field key={`${kebabCase(name)}-field`} name={name}>
-            {({ input: { value, onChange }, meta: { pristine, error } }) => {
-              const key = kebabCase(name);
-              const type = name in types ? types[name] : 'text';
-              const placeholder =
-                name in placeholders ? placeholders[name] : '';
-              return (
-                <Col key={key} xs={12}>
-                  <FormGroup>
-                    {type === 'hidden' ? null : (
-                      <ControlLabel htmlFor={key}>{label}</ControlLabel>
-                    )}
-                    <FormControl
-                      componentClass={type === 'textarea' ? type : 'input'}
-                      id={key}
-                      name={name}
-                      onChange={onChange}
-                      placeholder={placeholder}
-                      required={required.includes(name)}
-                      rows={4}
-                      type={type}
-                      value={value}
-                    />
-                    {error && !pristine ? (
-                      <HelpBlock>
-                        <Alert bsStyle='danger'>{error}</Alert>
-                      </HelpBlock>
-                    ) : null}
-                  </FormGroup>
-                </Col>
-              );
-            }}
-          </Field>
-        ))}
-    </div>
+    <SolutionFormContext.Consumer>
+      {({ displayErrors }) => {
```

Comment:
The displayErrors state does not seem to be propagating.

Since we are checking and displaying errors from the validators in `nullOrWarning`, I was wondering if we would need to pass this state down here at all.

So `solutionForm` would be in charge of opening modal using the validation check in `Form`, and
`FormFields` would be in charge of showing the error since it validates on each input change.

It would be nice to find the way to run the validations once, but that could wait for another optimization pr.

Reply:
I am not sure I understand...
- The state appears to be behaving as expected.
- Do you mean for this logic: https://github.com/freeCodeCamp/freeCodeCamp/pull/40225/files#diff-0f43bd67cd3701ceb7d5e86576b27ac0988e55d3d34e43cfbef44d520911e102R28-R35
to be handled within `FormFields.js`?